### ISSUE TITLE
Negartive value during depositAll usage

### ISSUE THREAD

**Issue by me-ann1** [iron-fish/ironfish#2518]
### What happened?

Steps to reproduce:
1. run command: ironfish depositAll -f 5000 --confirm
2. See output "Confirmed:   $IRON -0.00167606" **<= bug**

Issue reproduce: sometime 
Version              0.1.51 @ src

### Version

v0.1.51 

### Debug output

```shell
ironfish debug
yarn run v1.22.19
$ yarn build && yarn start:js debug
$ tsc -b
$ cross-env OCLIF_TS_NODE=0 IRONFISH_DEBUG=1 node --expose-gc --inspect=:0 --inspect-publish-uid=http --enable-source-maps bin/run debug
Database in use, skipping output that requires database.
Stop the node and run the debug command again to show full output.

Iron Fish version               0.1.51 @ src
Iron Fish library               0.0.28 @ src
Operating system                Linux x64
CPU model(s)                    AMD Ryzen 7 3700X 8-Core Processor
CPU threads                     16
RAM total                       94.30 GiB
Heap total                      4.00 GiB
Node version                    v16.18.0
ironfish in PATH                true
Garbage Collector Exposed       true
Telemetry enabled               true
Node name                       best2
Block graffiti                  best2
```

### Relevant log output

```shell
Following output appear in some cases: 
Balance
Confirmed:   $IRON -0.00167606
Unconfirmed: $IRON 0.00000000
Pending:     $IRON 0.06030556

--- Completed Transactions (0) ---
```

### Graffiti

best2

**Comment by hughy** [comment]
@me-ann1 could you share the output of `ironfish accounts:balance --explain`?

**Comment by me-ann1** [comment]
ironfish accounts:balance --explain
yarn run v1.22.19
$ yarn build && yarn start:js accounts:balance --explain
$ tsc -b
$ cross-env OCLIF_TS_NODE=0 IRONFISH_DEBUG=1 node --expose-gc --inspect=:0 --inspect-publish-uid=http --enable-source-maps bin/run accounts:balance --explain
Account: best2

Your balance is made of notes on the chain that are safe to spend
Balance: $IRON 0.00162606

1 notes worth 0.01036547 are on the chain within 2 blocks of the head
Unconfirmed: 0.01199153

2 notes worth 0.04826403 are waiting for miners to add them to the chain
Pending: 0.06025556

**Comment by me-ann1** [comment]
@hughy sent current output. 

Or need to get such info exactly during bug reproduce?

**Comment by hughy** [comment]
This should be resolved by #2565 

Thanks for reporting!
